### PR TITLE
feat: Check login status for all pages

### LINKED ISSUE #70: feat: Check login status for all pages
Should not be able to access /history or /notifications or /settings directly from the address bar.

### PR DESCRIPTION
**Pull request by marimarimarietty**
need to change status for settings page later.

### CODE REVIEW

File: src/pages/history.tsx [hunk 1]
```diff
@@ -1,13 +1,28 @@
...
 export default function HistoryIndex() {
   const pageid = "history";
 
-  return (
-    <PageTemplate pageid={pageid}>
-      <History />
-    </PageTemplate>
-  );
+  const [page, setPage] = useState<any>();
+  const [isLoggedin, setLoggedin] = useRecoilState<boolean>(loginState);
```
Comment (Takashicc) [inline src/pages/history.tsx]: Use `useRecoilValue`. Because we just want the value. we don't need setter.

File: src/pages/index.tsx [hunk 2]
```diff
@@ -4,22 +4,21 @@ import PageTemplate from "@/components/templates/PageTemplate";
...
+import { useRecoilState } from "recoil";
+import { loginState } from "@/globalStates/atoms/Auth";
 
 export default function Index() {
   const pageid = "index";
-  const [isLoggedin, setLoggedin] = useState<boolean>(
-    Cookies.get("access_token") ? true : false
-  );
   const [page, setPage] = useState<any>();
+  const [isLoggedin, setLoggedin] = useRecoilState<boolean>(loginState);
```
Comment (Takashicc) [inline src/pages/index.tsx:13]: Use `useRecoilValue`

File: src/components/organisms/Header.tsx [hunk 3]
```diff
@@ -5,14 +5,13 @@ import { SlideDown } from "react-slidedown";
...
 
-type Props = {
-  setLoggedin: React.Dispatch<React.SetStateAction<boolean>>;
-};
-
-export default function Header({ setLoggedin }: Props) {
+export default function Header() {
   const [width, setWidth] = useState<number>(0);
   const [openMenu, setOpenMenu] = useState<boolean>(false);
+  const [isLoggedin, setLoggedin] = useRecoilState<boolean>(loginState);
```
Comment (Takashicc) [inline src/components/organisms/Header.tsx]: Use `useSetRecoilState`. `isLoggedin` isn't used. just setting.

File: src/components/molecules/Signup.tsx [hunk 4]
```diff
@@ -157,6 +155,7 @@ export default function Form({ setLoggedin }: Props) {
   } = useForm<SignupFormInput>();
   const [isChecked, setChecked] = useState<boolean>(false);
   const [errMessage, setErrMessage] = useState<string | null>();
+  const [isLoggedin, setLoggedin] = useRecoilState<boolean>(loginState);
```
Comment (Takashicc) [inline src/components/molecules/Signup.tsx]: Use `useSetRecoilState`. `isLoggedin` isn't used. just setting.

File: src/components/molecules/Login.tsx [hunk 5]
```diff
@@ -100,13 +98,14 @@ const ErrorMsg = styled("p", {
...
 
-export default function Login({ setLoggedin }: Props) {
+export default function Login() {
   const {
     register,
     handleSubmit,
     formState: { errors },
   } = useForm<LoginFormInput>();
   const [errMessage, setErrMessage] = useState<string | null>();
+  const [isLoggedin, setLoggedin] = useRecoilState<boolean>(loginState);
```
Comment (Takashicc) [inline src/components/molecules/Login.tsx]: Use `useSetRecoilState`. `isLoggedin` isn't used. just setting.

File: src/pages/notifications.tsx [hunk 6]
```diff
@@ -1,13 +1,28 @@
...
   const pageid = "notifications";
 
-  return (
-    <PageTemplate pageid={pageid}>
-      <Notifications />
-    </PageTemplate>
-  );
+  const [page, setPage] = useState<any>();
+  const [isLoggedin, setLoggedin] = useRecoilState<boolean>(loginState);
+  const router = useRouter();
```
Comment (Takashicc) [inline src/pages/notifications.tsx]: Use `useRecoilValue`. `setLoggedin` is not been used.

**Review by Takashicc — CHANGES_REQUESTED** [review]

File: src/pages/index.tsx (hunk 2 above)
Comment (marimarimarietty) [reply src/pages/index.tsx:13]: I also used setLoggedin, so I didn't change it.

**Review by Takashicc — APPROVED** [review]

**Pull request merged**